File: A's/cpp/144A.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int n; cin >> n;
    vector<int> s(n);
    for (int i = 0; i < n; i++) {
        cin >> s[i];
    }

    int mi = s[0], ma = s[0];
    int mi_i = 0, ma_i = 0;
    for (int i = 0; i < n; i++) {
        mi = min(s[i], mi);
        ma = max(s[i], ma);
    }

    // Find the leftmost occurrence of the maximum value
    for (int i = 0; i < n; i++) {
        if (s[i] == ma) {
            ma_i = i;
            break; // Leftmost occurrence
        }
    }

    // Find the rightmost occurrence of the minimum value
    for (int i = n - 1; i >= 0; i--) {
        if (s[i] == mi) {
            mi_i = i;
            break; // Rightmost occurrence
        }
    }

    // Calculate the total seconds
    int secs = ma_i + (n - 1 - mi_i);
    if (mi_i < ma_i) {
        secs--; // Subtract 1 if they cross each other
    }
    cout << secs << endl;
    
    return 0;
}

File: A's/py/144A.py
def main() -> None:
    n: int = int(input())
    s: list[int] = list(map(int, input().split()))
    ma: int = max(s)
    mi: int = min(s)

    ma_i: int = 0
    mi_i: int = 0
    for i in range(n):
        if s[i] == ma: ma_i = i
        elif s[i] == mi: mi_i = i
    
    secs: int = 0
    if mi_i < ma_i:  # If min height is before the max height
        secs = ma_i + (n - mi_i) - 2
    else:
        secs = ma_i + (n - mi_i)
    
    print(secs)

if __name__ == "__main__":
    main()